subsys/bluetooth/host/hci_core.c

```diff
@@ -5004,6 +4998,22 @@ int bt_configure_data_path(uint8_t dir, uint8_t id, uint8_t vs_config_len,
 /* Return `true` if a command was processed/sent */
 static bool process_pending_cmd(k_timeout_t timeout)
 {
+	if (!atomic_test_bit(bt_dev.flags, BT_DEV_OPEN)) {
+		struct net_buf *buf;
+
+		LOG_WRN("Dropping queued commands since HCI transport is closed");
+
+		while ((buf = k_fifo_get(&bt_dev.cmd_tx_queue, K_NO_WAIT))) {
+			if (cmd(buf)->sync) {
+				cmd(buf)->status = BT_HCI_ERR_UNSPECIFIED;
+				k_sem_give(cmd(buf)->sync);
+			}
+			net_buf_unref(buf);
+		}
+
+		return false;
+	}
```

Comment:
Would it make sense to perform this action as part of `bt_disable` instead?

Reply:
> Perhaps we need something analogous to the `READY` flag for the `bt_disable()` operation, i.e. something that indicates that "now I'm really done". However, this is starting to smell a bit like scope creep, since this issue exists already today.

Agreed; although given the title of this PR, this seems like the main thing this PC is attempting to fix, which it seemingly doesn't

Reply:
I think I want the opposite direction of PR. Let's "over-engineer" it to be super-smooth. I think all we need is this:

- A flag that selects current target state "BT_WANT_ENABLED". This flag can be flipped at any time.
- A state machine with four states "DISABLED, ENABLING, ENABLED, DISABLING". It transitions from DISABLED to ENABLING whenever BT_WANT_ENABLED is set, and it transitions from ENABLED to DISABLING whenever BT_WANT_ENABLED is not set.

There can be sub-states of ENABLING and DISABLING if we want to break that up into smaller transitions.

Reply:
This entire thing started with me wanting appropriate protections for the public HCI command sending API, prompting me to at the very least want a flag to indicate if from an HCI driver API perspective the HCI transport is available. As long as we can accurately track that, I'm open to discussing whatever other flags we need.